src/pages/Swap/index.tsx

```diff
@@ -1,8 +1,236 @@
+  PrimaryButton,
+  SelectTokenTextField,
+  TertiaryButton,
+} from 'components';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'translation';
+import { Token, TokenId } from 'types';
+import { convertWeiToTokens, formatToReadablePercentage, getToken } from 'utilities';
 
-const SwapUi: React.FC = () => <></>;
+import { TOKENS } from 'constants/tokens';
 
-const Swap: React.FC = () => <SwapUi />;
+import { useStyles } from './styles';
+import { Swap } from './types';
+import useGetSwapInfo from './useGetSwapInfo';
 
-export default Swap;
+const tokenIds = Object.keys(TOKENS) as TokenId[];
+
+const SLIPPAGE_TOLERANCE_PERCENTAGE = 0.5;
+const readableSlippageTolerancePercentage = formatToReadablePercentage(
+  SLIPPAGE_TOLERANCE_PERCENTAGE,
+);
+
+interface FormValues {
+  fromToken: Token;
+  fromTokenAmountTokens: string;
+  toToken: Token;
+  toTokenAmountTokens: string;
+  direction: 'exactAmountIn' | 'exactAmountOut';
+}
+
+const initialFormValues: FormValues = {
+  fromToken: getToken('bnb'),
+  fromTokenAmountTokens: '',
+  toToken: getToken('xvs'),
+  toTokenAmountTokens: '',
+  direction: 'exactAmountIn',
+};
+
+export interface SwapPageUiProps {
+  formValues: FormValues;
+  setFormValues: (setter: (currentFormValues: FormValues) => FormValues) => void;
+  swapInfo?: Swap;
+}
+
+const SwapPageUi: React.FC<SwapPageUiProps> = ({ formValues, setFormValues, swapInfo }) => {
+  const styles = useStyles();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    // Reinitialize form values if swap becomes invalid
+    if (!swapInfo) {
+      setFormValues(currentFormValues => ({
+        ...currentFormValues,
+        fromTokenAmountTokens: '',
+        toTokenAmountTokens: '',
+      }));
+    }
+
+    if (swapInfo?.direction === 'exactAmountIn') {
+      setFormValues(currentFormValues => ({
+        ...currentFormValues,
+        toTokenAmountTokens: convertWeiToTokens({
+          valueWei: swapInfo.expectedToTokenAmountReceivedWei,
+          tokenId: swapInfo.toToken.id,
+        }).toFixed(),
+      }));
+    }
+
+    if (swapInfo?.direction === 'exactAmountOut') {
+      setFormValues(currentFormValues => ({
+        ...currentFormValues,
+        fromTokenAmountTokens: convertWeiToTokens({
+          valueWei: swapInfo.expectedFromTokenAmountSoldWei,
+          tokenId: swapInfo.fromToken.id,
+        }).toFixed(),
+      }));
+    }
+  }, [swapInfo]);
+
+  const switchTokens = () =>
+    setFormValues(currentFormValues => ({
+      ...currentFormValues,
+      fromToken: currentFormValues.toToken,
+      fromTokenAmountTokens: currentFormValues.toTokenAmountTokens,
+      toToken: currentFormValues.fromToken,
+      toTokenAmountTokens: currentFormValues.fromTokenAmountTokens,
+      direction:
+        currentFormValues.direction === 'exactAmountIn' ? 'exactAmountOut' : 'exactAmountIn',
+    }));
+
+  return (
+    <Paper css={styles.container}>
+      <SelectTokenTextField
```

Comment:
Is there a reason we are not using our formik here?

Reply:
Ideally we'd use it, but unfortunately that would force us to store form values in two different ways because `Formik` only allows to access form values from within the `<Formik>` tag and in this case its validation schema and submit function are defined based on the form values

Reply:
You can handle both of those cases with formik
Validation schema - We have an example of a dynamic validation schema on the proposals form
Submit function - you can wrap the submit functions in another function with logic about how they are called

Do we need error messages on this form? I don't see any logic for that. Formik would handle that as well

Reply:
Validation schema: we do, but the value it uses to be defined dynamically does not come form the form itself, hence why it's doable.
Submit function: same issue here. Since we want to display the best trade possible to the user while they're completing the form, then using Formik means the logic for fetching and displaying the trade needs to be contained within the `Formik` tag. However, when submitting the form we then don't have access to this data since the `onSubmit` handler is attached to the `Formik` tag. The solution for that would be to store the trade data again in a state variable that sits outside of the `Formik` tag.

Using Formik here feels like a hack and will make the codebase harder to understand,